eip191/src/main/kotlin/org/kethereum/eip191/EIP191.kt

```diff
@@ -1,19 +1,45 @@
 package org.kethereum.eip191
 
-import org.kethereum.model.ECKeyPair
 import org.kethereum.crypto.signMessage
+import org.kethereum.crypto.signedMessageToKey
+import org.kethereum.crypto.toSignatureData
+import org.kethereum.model.ECKeyPair
+import org.kethereum.model.PublicKey
+import org.kethereum.model.SignatureData
+import org.komputing.khex.model.HexString
 
 /*
 *
 * http://eips.ethereum.org/EIPS/eip-191
 *
 */
 
+private const val MESSAGE_PREFIX: Byte = 0x19
+private const val PERSONAL_SIGN_VERSION: Byte = 0x45
 
 fun ECKeyPair.signWithEIP191(version: Byte, versionSpecificData: ByteArray, message: ByteArray) =
-        signMessage(0x19.toByte().toByteArray() + version.toByteArray() + versionSpecificData + message)
+    signMessage(fullMessage(version, versionSpecificData, message))
 
 fun ECKeyPair.signWithEIP191PersonalSign(message: ByteArray) =
-        signWithEIP191(0x45, ("thereum Signed Message:\n" + message.size).toByteArray(), message)
+    signWithEIP191(PERSONAL_SIGN_VERSION, personalSignVersionData(message), message)
+
+fun personallySignedMessageToKey(message: ByteArray, signature: String): PublicKey =
+    personallySignedMessageToKey(message, HexString(signature).toSignatureData())
+
+fun personallySignedMessageToKey(message: ByteArray, signature: SignatureData): PublicKey =
```

Comment:
let's rename to personallySignedMessageToPublicKey

Reply:
done